pkg/auth/auth.go

```diff
@@ -25,11 +27,30 @@ const (
 	oauthToken            = "oauth_token"
 )
 
-// TokenForHost retrieves an authentication token and the source of
-// that token for the specified host. The source can be either an
-// environment variable or from the configuration file.
+// TokenForHost retrieves an authentication token and the source of that token for the specified
+// host. The source can be either an environment variable, configuration file, or the system
+// keyring. In the latter case, this shells out to "gh config get" to obtain the token.
+//
 // Returns "", "default" if no applicable token is found.
 func TokenForHost(host string) (string, string) {
+	if token, source := TokenFromEnvOrConfig(host); token != "" {
+		return token, source
+	}
+
+	if ghExe, err := safeexec.LookPath("gh"); err == nil {
+		c := exec.Command(ghExe, "config", "get", "oauth_token", "--host", host)
```

Comment:
Is there a reason not to use `gh auth token --hostname`? Seems like this is the exact case we created that command.